Docs: update oauth app section (#126)
yanyongyu/githubkit: README.md

data: dict = github.graphql("{ viewer { login } }")
```

> ### Develop an OAuth APP/GitHub APP  with web flow

Reviewer: ```suggestion
### Develop an OAuth APP (GitHub APP) with web flow
```

access_token = auth.token
refresh_token = auth.refresh_token
# restore the user token from database

> # With  OAuthAPP

Reviewer: ```suggestion
# when using OAuth APP or GitHub APP without user-to-server token expiration
# you can use the access_token directly
```

user_github = github.with_auth(
    OAuthTokenAuthStrategy(
        "<client_id>", "<client_secret>", token=access_token 
    )
)

> # OR  with  GithubAPP

Reviewer: ```suggestion
# OR when using GitHub APP with user-to-server token expiration
# you can use the refresh_token to generate a new token
```

# you can use the refresh_token to generate a new token
user_github = github.with_auth(
    OAuthTokenAuthStrategy(

>         "<client_id>", "<client_secret>", refresh_token=refresh_token 

Reviewer: ```suggestion
        "<client_id>", "<client_secret>", refresh_token=refresh_token
```

)

# now you can act as the user
resp = user_github.rest.users.get_authenticated()
user: PublicUser | PrivateUser = resp.parsed_data

> 
> # you can get public email as
> email = user.email
> 

Reviewer: ```suggestion

# you can get the user login id now
login_id = user.login
```

Author: Thank you for review.. I will make changes accordingly..

Author: > 

login_id = user.login  return login username , Is that we mean to say
login_id = user.id return Id , but for purpose this will be used ?

Reviewer: user.login is the actual id for user. the user.id is the node id and has no meaning.

Reviewer: @BhuwanPandey i have made a mistake on this. `user.id` is the actual id for user and the `user.node_id` is the node id actually.

access_token = auth.token
refresh_token = auth.refresh_token
# restore the user token from database

# when using OAuth APP or GitHub APP without user-to-server token expiration
user_github = github.with_auth(
    OAuthTokenAuthStrategy(
        "<client_id>", "<client_secret>", token=access_token 
    )
)
# OR when using GitHub APP with user-to-server token expiration
# you can use the refresh_token to generate a new token
user_github = github.with_auth(
    OAuthTokenAuthStrategy(
        "<client_id>", "<client_secret>", refresh_token=refresh_token 
    )
)

# now you can act as the user
resp = user_github.rest.users.get_authenticated()
user: PublicUser | PrivateUser = resp.parsed_data

> 
> # you can get the user login id now
> login_id = user.login
> 

Reviewer: ```suggestion

# you can get the user login id now
login_id = user.login
```

> 
> # you can get the user login id now
> login_id = user.login
> 
> ```

Reviewer: ````suggestion

# you can get the user login id now
login_id = user.login
```
````

access_token = auth.token
refresh_token = auth.refresh_token
# restore the user token from database

# when using OAuth APP or GitHub APP without user-to-server token expiration
user_github = github.with_auth(
    OAuthTokenAuthStrategy(

>         "<client_id>", "<client_secret>", token=access_token 

Reviewer: ```suggestion
        "<client_id>", "<client_secret>", token=access_token
```
